sdk/go/aws/ec2/pulumiTypes.go

```diff
@@ -14632,8 +14689,8 @@ func (o GetLaunchTemplateMonitoringArrayOutput) Index(i pulumi.IntInput) GetLaun
 }
 
 type GetLaunchTemplateNetworkInterface struct {
-	AssociatePublicIpAddress *bool `pulumi:"associatePublicIpAddress"`
-	DeleteOnTermination      bool  `pulumi:"deleteOnTermination"`
+	AssociatePublicIpAddress *bool  `pulumi:"associatePublicIpAddress"`
+	DeleteOnTermination      string `pulumi:"deleteOnTermination"`
```

Comment:
(Also left a comment on the TS side)

Reply:
This is not - we need to override this as its due to their "nullable bool" approach